torch/testing/_internal/distributed/rpc/jit/rpc_test.py

```diff
@@ -113,6 +145,61 @@ def rref_script_annotation(rref_var):
     return rref_python_annotation(rref_var).to_here()
 
 
+class LocalRRefTest(RpcAgentTestFixture):
+    @dist_init
+    def test_create_local_script_class_rref(self):
+        if self.rank != 0:
+            return
+
+        # Create a local RRef<MyScriptClass>.
+        rref_script_class = rpc.RRef(MyScriptClass(self.rank, ))
+        ret = rref_script_class.to_here().get_value()
+        self.assertEqual(ret, self.rank)
+
+    @dist_init
+    def test_create_local_script_module_rref(self):
+        if self.rank != 0:
+            return
+
+        # Create a local RRef<MyModuleInterface>.
+        rref_script_module = rpc.RRef(MyScriptModule(self.rank), MyModuleInterface) 
+        ret = rref_script_module.to_here().forward()
+        self.assertEqual(ret, torch.ones(self.rank))
+
+        # Create a local RRef<MyModuleInterface> without type hint.
+        with self.assertRaisesRegex(
+            RuntimeError, (
+                "The RRef being created contains a ScriptModule, "
+                "must provide it's ModuleInterface type hint."
+            )
+        ):
+            rref_script_module = rpc.RRef(MyScriptModule(self.rank))
+
+    @dist_init
+    def test_return_local_script_module_rref_in_py_and_use_in_script(self):
+        if self.rank != 0:
+            return
+
+        dst_worker_name = "worker{}".format((self.rank + 1) % self.world_size)
+
+        # Create a local RRef<MyModuleInterface> in Python
+        rref = rpc.rpc_sync(dst_worker_name, owner_create_rref_my_script_module, args=(self.rank,))
+        # Use RRef<MyModuleInterface> remotely in Script.
+        ret = rpc.rpc_sync(
+            rref.owner(), script_run_forward_rref_my_script_module, args=(rref,)
+        )
+        self.assertEqual(ret, torch.ones(self.rank))
+
+        # Use RRef<MyModuleInterface> remotely in Script with wrong argument type hint.
+        fut = rpc.rpc_async(
+            rref.owner(), script_run_forward_rref_my_script_module_2, args=(rref,)
```

Comment:
Just want to check if I understand this correctly. This fails because it expects a `MyModuleInterface2`, but the RRef actually holds a `MyModuleInterface`?

Reply:
Yes.

Reply:
why we need to do this check? I think this is testing the jit function type matching for module interface which already been covered somewhere else.

Reply:
@wanchaol Good point. I think it also add burdens if one day JIT framework decide to throw a different error message. They would be astonished to see this RPC test is broken. Let me remove it.